api/app/controllers/mail_controller.rb

```diff
@@ -0,0 +1,24 @@
+class MailController < ApplicationController  
```

Comment:
I think it would make more a lot more sense to have this as part of the `PlansController`. `send_request_approval_email` could be like `request_approval`.

Also do we use `send_plan_approved_email`

Reply:
Also I think we should add a field onto `Plan` called something like `is_requesting_approval`, so on the advisor side we can show the appropriate alerts/text.

Reply:
Added a field that stores the timestamp of when an approval is requested. We reset the nil when the plan is approved. By using a timestamp instead of a boolean, we can do things like throttle approval requests etc...